activestorage/lib/active_storage/attached/many.rb

```diff
@@ -9,7 +9,7 @@ class Attached::Many < Attached
     #
     # All methods called on this proxy object that aren't listed here will automatically be delegated to +attachments+.
     def attachments
-      record.public_send("#{name}_attachments")
+      change.present? ? change.attachments : record.public_send("#{name}_attachments")
```

Comment:
this either returns an `Array` or `ActiveStorage::Attachment::ActiveRecord_Associations_CollectionProxy` which makes the interface quite a moving target. with the latter i can do something like `attachments.where(...)` whereas that breaks when getting an array.